src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3405,6 +3408,41 @@ void ApplicationManagerImpl::OnHMILevelChanged(
     mobile_apis::HMILevel::eType from,
     mobile_apis::HMILevel::eType to) {
   LOG4CXX_AUTO_TRACE(logger_);
+  ProcessPostponedMessages(app_id);
+  ProcessApp(app_id, from, to);
+}
+
+void ApplicationManagerImpl::ProcessPostponedMessages(const uint32_t app_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ApplicationSharedPtr app = application(app_id);
+  if (!app) {
+    LOG4CXX_WARN(logger_, "The app with id: " << app_id << " does not exist");
+    return;
+  }
+  MobileMessageQueue messages;
+  app->SwapMobileMessageQueue(messages);
+  auto push_allowed_messages =
+      [this, &app](smart_objects::SmartObjectSPtr message) {
+        const std::string function_id = MessageHelper::StringifiedFunctionID(
+            static_cast<mobile_apis::FunctionID::eType>(
+                (*message)[strings::params][strings::function_id].asUInt()));
+        const RPCParams params;
+        const mobile_apis::Result::eType check_result =
+            CheckPolicyPermissions(app, function_id, params);
+        if (check_result == mobile_api::Result::SUCCESS) {
```

Comment:
@LitvinenkoIra I would recommend to swap variables in comparison

Reply:
@AByzhynar Done e638bda5ff050f5c1d32aeb2dc575c1016170ca1